common/djangoapps/student/views.py

```diff
@@ -1384,7 +1384,9 @@ def login_user(request, error=""):  # pylint: disable=too-many-statements,unused
             user = User.objects.get(email=email)
         except User.DoesNotExist:
             try:
-                user = User.objects.get(username=email)
+                # try using the email as username
+                username = email
```

Comment:
Well... I know I said `username=email` isn't a good idea but neither is `username = email`... :p